metaflowbot/metaflowbot/actions/runs/commands.py

```diff
@@ -0,0 +1,151 @@
+                                                     HEADINGS,
+                                                     SLACK_MAX_BLOCKS,
+                                                     error_message)
+from metaflowbot.state import MFBState
+
+from .run_resolver import (ResolvedRun, RunNotFound, RunResolver,
+                           RunResolverException, datetime_response_parsing,
+                           find_user, step_runtime)
+
+
+@action.command(help="Set a new run to be inspected")
+@click.option('--runspec',
+              help="A query to find a run to inspect")
+@click.option('--howto/--no-howto',
+              help="Only show help text")
+@click.pass_context
+def inspect_run(ctx, runspec=None, howto=False):
+    obj = ctx.obj
+    resolver = RunResolver('inspect')
+    if howto:
+        obj.reply(howto_inspect_run(resolver))
+    else:
+        try:
+            obj.reply("Searching runs. Just a minute...")
+            runs = resolver.resolve(runspec)
+            if len(runs) == 1:
+                attrs = {'inspect.run_id': runs[0].id}
+                state = MFBState.message_set_attributes(obj.thread, attrs)
+                obj.publish_state(state)
+                obj.reply("Ok, inspecting *%s*." % (runs[0].id))
+                ctx.invoke(inspect, run_id=runs[0].id)
+            else:
+                reply = resolver.format_runs(runs, lambda _: None)
+                obj.reply(reply)
+        except RunResolverException as ex:
+            obj.reply(str(ex))
+        except RunNotFound as ex:
+            obj.reply(str(ex))
+        except Exception as e:
+            traceback.print_exc()
+            my_traceback = traceback.format_exc()
+            obj.reply(DEFAULT_ERROR_MESSAGE,**error_message(my_traceback))
+
+@action.command(help="Inspect the current run or show help text")
+@click.option('--run-id',
+              help="Run ID to inspect")
+@click.option('--create-thread/--no-create-thread',
+              help="Will create a new thread")
+@click.option('--howto/--no-howto',
+              help="Only show help text")
+@click.pass_obj
+def inspect(obj, run_id=None,create_thread=False, howto=False):
+    resolver = RunResolver('inspect')
+    if create_thread:
+        obj.publish_state(MFBState.message_new_thread(obj.thread))
+    if howto:
+        obj.reply(howto_inspect_run(resolver))
+    else:
+        try:
+            reply_inspect(obj, run_id)
+        except MetaflowNotFound as e:
+            obj.reply(HEADINGS.NO_RUNS.value)
+        except:
+            traceback.print_exc()
+            my_traceback = traceback.format_exc()
+            obj.reply(DEFAULT_ERROR_MESSAGE,**error_message(my_traceback))
+
+
+def run_status(run):
+    if run.finished:
+        if run.successful:
+            mins = datetime_response_parsing((DATEPARSER(run.finished_at) - DATEPARSER(run.created_at)).total_seconds())
+            return "It ran for %s and finished successfully." % mins
```

Comment:
For runs runnings for more than a week, we should emit - `..ran for x weeks and y days`, for more than a day - `..ran for x days and y hours`, for more than an hour - `..ran for x hours and y mins`, otherwise `..ran for x minutes` (with special attention to when to use singular vs plurals)

Reply:
`datetime_response_parsing` makes this string of `x weeks and y days`, I will fix this function to manage singular vs plurals better.

Reply:
Will also improve variable name here as it's confusing when glancing the function.

Reply:
New commits introduce this change.